Assets/Scripts/Logic/AIPlayer.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace johnny.HexProject.Logic
+{
+    public class AIPlayer : IPlayer
+    {
+        private static int _nextId = 0;
+        private readonly int _movementsPerStep;
+        private readonly int _stepsPerAttack;
+        private readonly int _attackRange;
+        private readonly IBoard _board;
+        private Tile _tile;
+        
+        public event Action PlayerDied;
+        public bool IsAlive => Health > 0;
+
+        public Tile Tile 
+        {
+            get => _tile;
+            set
+            {
+                if (_tile != null)
+                {
+                    _tile.IsOccupied = false;
+                }
+                _tile = value;
+                _tile.IsOccupied = true;
+            }
+        }
+        
+        private int _stepsCount;
+        private int _lastAttackStep;
+        private int _stepsInCurrentAttack;
+        public int Id { get; set; }
+        public int Health { get;  set; }
+
+        private AIPlayer() { }
+        public AIPlayer(int movementsPerStep, int stepsPerAttack,
+            int attackRange, int health, Tile startingTile, IBoard board)
+        {
+            Id = _nextId++;
+            _movementsPerStep = movementsPerStep;
+            _stepsPerAttack = stepsPerAttack;
+            _attackRange = attackRange;
+            Health = health;
+            Tile = startingTile;
+            _board = board;
+            
+            PlayerDied += () => Debug.Log($"Player {Id} died");
+        }
+
+        public void TakeDamage(int damage)
+        {
+            if (!IsAlive)
+                return;
+            Health -= damage;
+            if(!IsAlive)
+                PlayerDied?.Invoke();
+        }
+
+        public IReadOnlyList<ICommand> Step()
+        {
+            var commands = new List<ICommand>();
+            
+            if (!IsAlive)
+                return commands;
+            
+            _stepsCount++;
+            var opponent = PlayerManager.Instance.GetOpponent(this);
```

Comment:
Some optimizations could be considered for finding opponents. If a player A found his opponent is player B, it possibly means B's opponent is also A. It's not a rule but at least `distance(A, B)` serves as an upperbound for the distance that B needs to look to find an opponent, in case it haven't found yet. Opponents could also be stored as fields of the players, since once they are found, the are kept until one of them is destroyed and it wouldn't be necessary to keep looking for opponents in the next steps.